FRAuth/FRAuth/SocialLogin/AppleSignInHandler.swift

```diff
@@ -115,12 +125,19 @@ extension AppleSignInHandler: ASAuthorizationControllerDelegate {
         switch authorization.credential {
         case let appleIDCredential as ASAuthorizationAppleIDCredential:
             FRLog.v("ASAuthorizationAppleIDCredential received: \(appleIDCredential)")
-            guard let tokenData = appleIDCredential.identityToken, let idToken = String(data: tokenData, encoding: .utf8) else {
+                      
+            let appleSignInResponse = AppleSignInResponse(appleIDCredential)
+            guard let authorization_code = appleSignInResponse.code, let id_token = appleSignInResponse.id_token, let IDToken1tokenJSON = try? JSONEncoder().encode(appleSignInResponse), let IDToken1token = String(data: IDToken1tokenJSON, encoding: .utf8) else {
                 self.completionCallback?(nil, nil, SocialLoginError.unsupportedCredentials("Failed to parse received credentials data (ASAuthorizationAppleIDCredential.identityToken)"))
                 return
             }
             
-            self.completionCallback?(idToken, self.tokenType, nil)
+            if self.acceptsJSON == true {
+                self.completionCallback?(IDToken1token, self.tokenType, nil)
+            } else {
+                self.completionCallback?(id_token, self.tokenType, nil)
+            }
+            
```

Comment:
either way works , but i am thinking of like this.   self.completionCallback?(self.acceptsJSON == true ? IDToken1token : id_token , self.tokenType, nil)

Reply:
I personally prefer the verbosity of the existing solution as it makes it easier to test